### ISSUE TITLE
Bracket orders

### ISSUE THREAD

**Issue by Aalv3** [MicroTrendsLtd/NinjaTrader8#99]
What would cause bracket orders not to deploy on a live account? Mind you, I have submitted hundreds of orders testing in sim, and they work every time.

**Comment by MicroTrendsTom** [comment]
So the entry order goes in + Position fills?
but no stops/exits?
Is this is a rithmic thing?
Is it showing a position in the UI?

any errors/debug
can you watch it run through a VStudio debugger?

**Comment by MicroTrendsTom** [comment]
try in Unsafe mode - so it misses all the checks etc

**Comment by Aalv3** [comment]
It is on a rithmic prop account. Every strategy I have that uses the ATS backend does the same thing. On sim, I can submit 200 orders and they work perfect with the brackets. On live, it will randomly enter a position with no brackets at all. Sometimes it works, but 70% of the time it does not. I do not have a cash account to try it on at the moment. Not sure how to run in unsafe mode but I will look into this

**Comment by MicroTrendsTom** [comment]
maybe its related to this #45

**Comment by Aalv3** [comment]
Appears to be the same issue. I have ran sidler's version and it does the same thing. Also, my log looks the same as the one he posted. I'm not sure if a solution has been discovered. Will be looking into it

**Comment by MicroTrendsTom** [comment]
On absence of NT actually fixing the issue at source.
A solution is to add  a "Rithmic" mode and then after an entry -
force an execution OnExec to mimic the normal workflow -when its all filled.
This would be for example a second after the entry has gone in  etc to call back if rithmic mode 
or to poll after a new position and then fire the placeStopLossTraget Workflow.  
not a 5monute job for sure.

Note
May get addressed in Q2 base do on current time & resources.
We would only use it for testing/optimization only not for production trading as the AFt8 supersedes all this tech.
Note:  You can try plug your code algo signal logic into the AFT8 AlgoTemplates posted in the discord group etc.

**Comment by Aalv3** [comment]
> Ok i have opened a rithmic  live account will test out at some point when i can come up for air

Thank you so much for the help, you rock!

**Comment by MicroTrendsTom** [comment]
latest update the rithmic account was turned down and as yet there is not time to look at this due to other projects - this is still open

**Comment by JonathanGVC** [comment]
@MicroTrendsTom If you're still looking for an account, try apex. If you email them, link the github, and explain your situation, they might just give you a coupon for a free dev account, they have that arrangement with the company I consult for.

Barring that, they routinely run 80% off deals for 1 month, which brings their 25k account down to 30 bucks for the month (USD). They support MCL, which means inside that month you can test quite a bit before running into their $1500 trailing drawdown.

(My browser AV extension has a problem with the link, but it's legit, feel free to google it instead)
https://apextraderfunding.com/

All of what I said above is also true of LeeLoo, so you might want to try them as well.

**Comment by MicroTrendsTom** [comment]
> @MicroTrendsTom If you're still looking for an account, try apex. If you email them, link the github, and explain your situation, they might just give you a coupon for a free dev account, they have that arrangement with the company I consult for.
> 
> Barring that, they routinely run 80% off deals for 1 month, which brings their 25k account down to 30 bucks for the month (USD). They support MCL, which means inside that month you can test quite a bit before running into their $1500 trailing drawdown.
> 
> (My browser AV extension has a problem with the link, but it's legit, feel free to google it instead) https://apextraderfunding.com/
> 
> All of what I said above is also true of LeeLoo, so you might want to try them as well.
Yup out of these so called "prop firms" - some better than others Apex seems to be least finickity of the bunch and more popular. Here is a list here of the ones i find the better for NT :  https://algotradingsystems.net/GetStarted

The rithmic got turned down as im already using CQG in my retail account, so it would require a new account with a different brokerage/fcm to get that,.     https://algotradingsystems.net/GetStarted

The problem i have is time management  in 2023 there is no slots.
This product here https://algofuturestrader.com/   allows coding of algos far superior to this product for free but no 
backtesting - no issues with any broker as the workflow doesnt use strategies nut instead goes via the NT account api  and is broker neutral.  i might ahve to add a section in github so show how this is done in fact  i do show a bit in the trading group.
